src/controllers/createpdf.controller.ts

```diff
@@ -0,0 +1,92 @@
+import { Controller, Get, Res, Post, Body, Render, Param, Query} from '@nestjs/common';
+import {Response} from 'express';
+import {createUserData, createLogInfo,createRunConf} from '../dtos/createpdf.dto';
+import {CreatePdfService} from "../services/createpdf.service";
+import { Log } from '../entities/log.entity';
+import { getManager } from 'typeorm';
+import { Run } from '../entities/run.entity';
+import { ApiUseTags, ApiBearerAuth } from '@nestjs/swagger';
+
+@ApiUseTags('createpdf')
+@Controller('createpdf')
+export class CreatepdfController {
+  constructor(public createPdfService: CreatePdfService){};
+
+  @Get('getlog')
+  async getLogInfo(@Query() query: createLogInfo, @Res() res: Response): Promise<void>{
+    var allDate = this.createPdfService.getDate();
+
+    const repository = getManager().getRepository(Log);
```

Comment:
You should not talk to the database layer directly in the controller, use a service instead. Please extract all the code from row 19 until row 24 to a service instead.